### PR TITLE
array constructors

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: svd-rs/src/clusterinfo.rs [hunk 1]
```diff
@@ -146,7 +147,14 @@ impl ClusterInfo {
     pub fn builder() -> ClusterInfoBuilder {
         ClusterInfoBuilder::default()
     }
-
+    /// Construct single `Cluster`
```
Comment (Emilgardis) [inline svd-rs/src/clusterinfo.rs:150]: This, and the other instances, should maybe be an intra-doc link, feels kinda redundant but I usually do it anyway myself

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: svd-rs/src/clusterinfo.rs (hunk 1 above)
Comment (burrbull) [reply svd-rs/src/clusterinfo.rs:150]: square brackets?

File: svd-rs/src/registerinfo.rs [hunk 2]
```diff
@@ -229,11 +229,11 @@ impl RegisterInfo {
         RegisterInfoBuilder::default()
     }
     /// Construct single `Register`
-    pub fn single(self) -> Register {
+    pub const fn single(self) -> Register {
```
Comment (Emilgardis) [inline svd-rs/src/registerinfo.rs:232]: :D

**Review by Emilgardis — CHANGES_REQUESTED** [review]
lgtm, squash all commits

**Comment by burrbull** [comment]
done

**Review by Emilgardis — APPROVED** [review]
bors r+

some good usability and future compatibility :)

File: svd-rs/src/cluster.rs [hunk 3]
```diff
@@ -70,9 +81,9 @@ mod ser_de {
         {
             let ClusterArray { dim, info } = ClusterArray::deserialize(deserializer)?;
             if let Some(dim) = dim {
-                Ok(Cluster::Array(info, dim))
+                Ok(Self::Array(info, dim))
```
Comment (Emilgardis) [inline svd-rs/src/cluster.rs:84]: Why aren't these using the new functions?

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: svd-rs/src/cluster.rs (hunk 3 above)
Comment (burrbull) [reply svd-rs/src/cluster.rs]: done

**Review by Emilgardis — APPROVED** [review]
bors r+

**Pull request merged**